src/components/taskDetails/index.tsx

```diff
@@ -0,0 +1,228 @@
+import React, {
+  ChangeEvent,
+  FC,
+  useEffect,
+  useContext,
+  useReducer,
+  useRef,
+  useState,
+} from 'react';
+import useFetch from '@/hooks/useFetch';
```

Comment:
Use `redux-toolkit`, as it's used in this project.

Reply:
Mentioned this in above comment reply